Sources/NIOHTTP1/HTTPEncoder.swift

```diff
@@ -268,6 +268,10 @@ public final class HTTPResponseEncoder: ChannelOutboundHandler, RemovableChannel
                 buffer.write(response: response)
             }, context: context, headers: response.headers, promise: promise)
         case .body(let bodyPart):
+            if case let .byteBuffer(byteBuffer) = bodyPart, byteBuffer.readableBytes == 0 {
+                // empty body parts will end the response which we don't want
+                break
+            }
```

Comment:
Is this the right place to make this decision? I was expecting that we change this in the place where we currently send the `end` when we write an empty body part instead of holding back the write here.

Reply:
This happens in llhttp which is C code transpiled from TypeScript which we don't want to change.

Reply:
@FranzBusch done in https://github.com/apple/swift-nio/pull/2531/commits/abe6ae2c22454fc01b03667cb1ac84ffed9e5c2f